lib/active_admin/base_controller/authorization.rb

```diff
@@ -100,7 +100,7 @@ def action_to_permission(action)
       end
 
       def dispatch_active_admin_access_denied(exception)
-        call_method_or_exec_proc active_admin_namespace.on_unauthorized_access, exception
+        instance_exec(self, exception, &active_admin_namespace.on_unauthorized_access.to_proc)
```

Comment:
This seems backwards incompatible. Before we would allow passing a string, like '"to_s"', now we don't, I think ("to_s".to_proc NoMethodError: undefined method `to_proc' for "to_s":String).

Also, I'm not sure about the reduction in complexity. I feel it's nice to keep all this magic in a single place.

Reply:
Yes.  Why anyone would pass a method name as string here I really don't know, but backward compatibility would be assured by adding 
```
setting = active_admin_namespace.on_unauthorized_access
setting = setting.to_sym if setting.is_a?(String)
instance_exec(... &setting.to_proc)
```

Reply:
I wouldn't really care about backwards compatibility here. I just mentioned it since the patch cares, so I assumed you had more context about external usage.